src/TinyGPS++.cpp

```diff
@@ -97,6 +97,16 @@ bool TinyGPSPlus::encode(char c)
   return false;
 }
 
+bool TinyGPSPlus::encode(char* c, size_t len)
+{
+  bool bRet = false;
+
+  for(int i = 0; i < len; i++)
```

Comment:
I think you should return on a complete sentence, so you can process the data first.
But this also complicates stuff a bit as you then also should keep track of how many characters were read and keep the rest.
Or... you should pre-parse and only hand over chunks till the next newline char.

Reply:
It would be a nice design change but that would also require several other changes in existing components of the library for your suggestion to integrate nicely. And as you mentioned already, that would make things more complicated. And even the way I have done, every statement gets processed, exactly like it would be by using the existing interface function which takes one character at a time (my change uses existing function anyway).

I had thought about what you have suggested but decided to implement this change the way I did because of the following reasons.

1. Simplicity of the code change and the library interface (the user code remains simple as well).

2. Consistency with the existing design and library interface. Existing encode function returns true upon completion of "A" sentence. The caller has no idea which sentence it was and what information (location? time? speed? etc.) was just updated as a result of this completion.

In a similar fashion, after my change, the caller knows that one or more valid sentences were successfully parsed by the library, but still does not know what was updated.

Therefore, this change is perfectly consistent with the existing library and it's external interface.

It would be another nice change if the library held "dirty" flags for all pieces of information it holds, so that the caller can figure out which fields contain new information.

3. Minimal changes to existing code while still achieving what I wanted to achieve.